File: 01 - Applications/futnetsinglemasterts/src/utils/beans/Rencontre.ts
export default class Rencontre {
  idRencontre: Number
  date: Date
  joueur1: Number
  joueur2: Number
  score1: Number
  score2: Number
  valide: boolean
  constructor(
    idRencontre: Number,
    date: Date,
    joueur1: Number,
    joueur2: Number,
    score1: Number,
    score2: Number,
    valide: boolean
  ) {
    this.idRencontre = idRencontre
    this.date = date
    this.joueur1 = joueur1
    this.joueur2 = joueur2
    this.score1 = score1
    this.score2 = score2
    this.valide = valide
  }
}

File: 01 - Applications/api/src/main/java/ch/futnetsinglemaster/api/entity/Rencontre.java
package ch.futnetsinglemaster.api.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@Entity
@Table(name = "rencontre")
public class Rencontre {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_rencontre", nullable = false)
    private Integer id;

    @NotNull
    @Column(name = "valide", nullable = false)
    private Byte valide;

    @NotNull
    @Column(name = "date", nullable = false)
    private LocalDate date;

}